### DISCUSSION TITLE
Timeout

### DISCUSSION THREAD

**Discussion by Skife114** [OndrejNepozitek/Edgar-Unity#122] [General]
i used this Package to create dungeons with the assets i had, and i used all the guilds that i had and first problem had was that door size wasn't the same so i fixed it. when i pressed the generate level, i had Timeout diagnostics because it said the following:
1) the generator was not able to produce level within a given time limit. the reason for that is usually that there is some problem with the configuration of the generator.
2) it seems like we were not able to automatically detect any problems with the configuration.

so i create GitHub post

**Comment by OndrejNepozitek** [comment]
Hello, I recommend starting small. Try to generate a level with 2-4 rooms and see if that works. Also, you can post some screenshots of your level graph and room templated - that might help diagnose the problem.

**Reply by Skife114** [reply]
thanks for the fast replay, il send hear all the files/photos of the project
![photo 3](https://github.com/OndrejNepozitek/Edgar-Unity/assets/80407515/7c2ad4f0-71ba-40b6-a959-ea0edd74a0ad)
![photo 4](https://github.com/OndrejNepozitek/Edgar-Unity/assets/80407515/0f6282d4-7927-486a-8717-90a91ba5f269)
![photo 7](https://github.com/OndrejNepozitek/Edgar-Unity/assets/80407515/034f50a7-5e23-4699-a627-79af5cd90962)
![photo 1](https://github.com/OndrejNepozitek/Edgar-Unity/assets/80407515/9aa9457c-0bbd-40bf-8767-676fa7db8f9f)
![photo 6](https://github.com/OndrejNepozitek/Edgar-Unity/assets/80407515/fd5b156f-eefd-4865-bd36-64bcf761c37e)
![photo 2](https://github.com/OndrejNepozitek/Edgar-Unity/assets/80407515/775d49a6-e833-497c-a9d9-76135a79fb5c)
![photo 5](https://github.com/OndrejNepozitek/Edgar-Unity/assets/80407515/14f99ced-b17d-4e24-a8eb-75cd5839fa61)

**Reply by Skife114** [reply]
![photo 8](https://github.com/OndrejNepozitek/Edgar-Unity/assets/80407515/84cdc442-2552-40d7-9c95-8bb485b7856e)

**Reply by OndrejNepozitek** [reply]
I think your setup is just impossible to generate. Your "Enemy" room has doors only on the left and bottom sides, but you want to connect 3 such rooms next to each other. That is not possible. You need to have some rooms with door on some other sides.

Look at your boss room. The door is on top. This boss room should be connect to Enemy room, which has door on bottom and left. So the bottom door is connected to the boss room and the left door is open. Now you need to connect another Enemy room to the previous room and the door needs to be on the right side of the room templates in order to connect it to the open door that faces to the left. And here is the problem - you don't have a single room template with a door facing to the right, meaning the generator can never generate such level.

So you either need rooms with more univeral setup of doors, ideally doors facing to all 4 directions. Or you need more room template with various combinations of doors.

**Reply by Skife114** [reply]
yes i see, and also sloved it with more templates and more logical connections between, the size isnt the issue but the frame/built
